controllers/authpolicy_controller_test.go

```diff
@@ -49,6 +49,28 @@ var _ = Describe("AuthPolicy controller", func() {
 
 	AfterEach(DeleteNamespaceCallback(&testNamespace))
 
+	apFactory := func(mutateFn func(policy *api.AuthPolicy)) *api.AuthPolicy {
+		policy := &api.AuthPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "toystore",
+				Namespace: testNamespace,
+			},
+			Spec: api.AuthPolicySpec{
+				TargetRef: gatewayapiv1alpha2.PolicyTargetReference{
+					Group:     "gateway.networking.k8s.io",
+					Kind:      "HTTPRoute",
+					Name:      testHTTPRouteName,
+					Namespace: ptr.To(gatewayapiv1.Namespace(testNamespace)),
+				},
+				AuthScheme: testBasicAuthScheme(),
+			},
+		}
+		if mutateFn != nil {
+			mutateFn(policy)
+		}
+		return policy
+	}
```

Comment:
```suggestion
	apFactory := func(mutateFns ...func(policy *api.AuthPolicy)) *api.AuthPolicy {
		policy := &api.AuthPolicy{
			ObjectMeta: metav1.ObjectMeta{
				Name:      "toystore",
				Namespace: testNamespace,
			},
			Spec: api.AuthPolicySpec{
				TargetRef: gatewayapiv1alpha2.PolicyTargetReference{
					Group:     "gateway.networking.k8s.io",
					Kind:      "HTTPRoute",
					Name:      testHTTPRouteName,
					Namespace: ptr.To(gatewayapiv1.Namespace(testNamespace)),
				},
				AuthScheme: testBasicAuthScheme(),
			},
		}
		for _, mutateFun := range mutateFns {
			mutateFn(policy)
		}
		return policy
	}
```

Just a very small optional suggestion. The only difference is that using this it's possible to call `apFactory()` instead of `apFactory(nil)` to get the default AuthPolicy